FWCore/Services/plugins/tracer_setupFile.cc

```diff
@@ -0,0 +1,814 @@
+#include "tracer_setupFile.h"
+#include "monitor_file_utilities.h"
+
+#include <chrono>
+
+#include <sstream>
+#include <type_traits>
+#include <cassert>
+#include <typeindex>
+
+#include "FWCore/Concurrency/interface/ThreadSafeOutputFileStream.h"
+
+#include "DataFormats/Provenance/interface/EventID.h"
+#include "DataFormats/Provenance/interface/LuminosityBlockID.h"
+#include "DataFormats/Provenance/interface/ModuleDescription.h"
+#include "DataFormats/Provenance/interface/RunID.h"
+#include "DataFormats/Provenance/interface/Timestamp.h"
+
+#include "FWCore/ServiceRegistry/interface/GlobalContext.h"
+#include "FWCore/ServiceRegistry/interface/ModuleCallingContext.h"
+#include "FWCore/ServiceRegistry/interface/ESModuleCallingContext.h"
+#include "FWCore/ServiceRegistry/interface/PathContext.h"
+#include "FWCore/ServiceRegistry/interface/ProcessContext.h"
+#include "FWCore/ServiceRegistry/interface/StreamContext.h"
+#include "FWCore/ServiceRegistry/interface/ActivityRegistry.h"
+
+#include "FWCore/Framework/interface/IOVSyncValue.h"
+#include "FWCore/Framework/interface/ESRecordsToProductResolverIndices.h"
+
+#include "FWCore/Utilities/interface/TimingServiceBase.h"
+
+using namespace edm::service::monitor_file_utilities;
+
+namespace {
+  using duration_t = std::chrono::microseconds;
+  using clock_t = std::chrono::steady_clock;
+  auto now = clock_t::now;
+
+  enum class step : char {
```

Comment:
Could this enum start with upper case letter, i.e. `Step`?

Reply:
done